### PR TITLE
[WIP] SATA (bundle saturation) QC scripts

### PR DESCRIPTION
**Pull request by frheault**
# Quick description

Scripts to evaluate saturation, will sample a bundle (or a tractogram) and see the impact at various streamline counts with random redraw. It outputs a json and a set of graph to evaluate if a bundle is full-enough
cc2_dice
cc2_entropy
cc2_picked
cc2_slope
cc2_volume
cc2_w_dice

## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Provide data, screenshots, command line to test (if relevant)

...

# Checklist

- [ ] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [ ] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

File: scripts/scil_evaluate_bundle_saturation.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+- Optionally, a whole-brain tractogram for comprehensive analysis
+    (`--whole_brain` argument).
+Both of these (if used) should be extremely dense tractograms, to ensure
+more than enough streamlines are available for sampling.
+
+Output:
+- A JSON file containing metrics such as volume, dice coefficient, entropy,
+    etc., computed at different sampling levels.
+- A set of PNG plots for each metric, illustrating how the metric values
+    change with different sampling levels.
```
Comment (mdesco) [inline scripts/scil_evaluate_bundle_saturation.py:23]: Should we explain these metric/plots?
I just ran the SATA script on a single bundle and I have to admit that I'm a bit lost in the different PNGs produced.

File: scripts/scil_evaluate_bundle_saturation.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+                         color='red', alpha=0.25)
+
+        for i, (x, y, s) in enumerate(zip(iteration_pick, avg, std)):
+            if i % 2 == 0:
+                plt.text(x*1.02, y*1.02,
+                         f"{y:.3f} +/- {s:.3f}", fontsize=4, ha='right')
+
+        # Set y-axis limits for specific measures
+        if measure in ['dice', 'w_dice', 'entropy']:
+            plt.ylim(0, 1.05)
```
Comment (mdesco) [inline scripts/scil_evaluate_bundle_saturation.py:263]: volume should probably start at 0

File: scripts/scil_evaluate_bundle_saturation.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+
+        plt.title(f"Measure: {measure}")
+        plt.xlabel("Iteration")
```
Comment (mdesco) [inline scripts/scil_evaluate_bundle_saturation.py:266]: Is Iteration the right word? 
# of streamlines could be better, no?

**Review by mdesco — APPROVED** [review]
Aside from the small comments, the rest is all good. Very useful.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Pull request closed without merging**